packages/flutter/lib/src/widgets/selectable_region.dart

```
@@ -2896,3 +3025,405 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
+    required this.selectionNotifier,
+    required this.child,
+  });
+
+  /// Notifies listeners when the selection has changed.
+  final SelectionListenerNotifier selectionNotifier;
+
+  /// The child widget this selection listener applies to.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  @override
+  State<SelectionListener> createState() => _SelectionListenerState();
+}
+
+class _SelectionListenerState extends State<SelectionListener> {
+  late final _SelectionListenerDelegate _selectionDelegate = _SelectionListenerDelegate(selectionNotifier: widget.selectionNotifier);
+  _SelectableRegionSelectionStatusNotifier? _selectableRegionScope;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _selectableRegionScope?.removeListener(_handleOnSelectableRegionChanged);
+    _selectableRegionScope = _SelectableRegionScope.maybeOf(context);
+    _selectableRegionScope?.addListener(_handleOnSelectableRegionChanged);
+  }
+
+  @override
+  void didUpdateWidget(SelectionListener oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.selectionNotifier != widget.selectionNotifier) {
+      _selectionDelegate.setNotifier(widget.selectionNotifier);
+    }
+  }
+
+  @override
+  void dispose() {
+    _selectionDelegate.dispose();
+    super.dispose();
+  }
+
+  void _handleOnSelectableRegionChanged() {
+    if (_selectableRegionScope != null) {
+      _selectionDelegate._dispatchStatusEvent(_selectableRegionScope!.selectionListenerStatus);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SelectionContainer(
+      delegate: _selectionDelegate,
+      child: widget.child,
+    );
+  }
+}
+
+class _SelectionListenerDelegate extends _SelectableRegionContainerDelegate {
+  _SelectionListenerDelegate({
+    required SelectionListenerNotifier selectionNotifier,
+  }) : _selectionNotifier = selectionNotifier {
+    _selectionNotifier._selectionDelegate = this;
```

What if the developer gives the same `SelectionListenerNotifier` to multiple `SelectionListener`s? The delegate would be overwritten right? And it seems it would be a silent failure?

---

We probably shouldn't allow the same `SelectionListenerNotifier` to be provided to multiple `SelectionListener`s. I added an assert, do you think it needs anything else?